crates/llm-base/src/inference_session.rs

```diff
@@ -897,8 +897,8 @@ pub fn feed_prompt_callback<'a, E: std::error::Error + 'static>(
 
 /// An [InferenceResponse] callback that will halt inference when a `stop_sequence` is generated.
 /// This callback is used in [InferenceSession::infer] in chat_mode.
-pub fn conversation_inference_callback<'a, E: std::error::Error + 'static>(
-    stop_sequence: String,
+pub fn conversation_inference_callback<'a, E: std::error::Error + Send + Sync + 'static>(
```

Comment:
what do these do?

Reply:
In Rust, objects get the `Send` trait if they can be sent across threads, and `Sync` if they can be used by multiple threads (you can see more details [here](https://doc.rust-lang.org/nomicon/send-and-sync.html)). 

I needed to add this because `eyre`, which we use for error reporting in the CLI, expects the error from `infer` to be `Send + Sync`. The error is passed down from `callback` to `infer`, so the trait requirements need to be updated across the library.